File: examples/javascript/sensortemplate.js
/*
 * The MIT License (MIT)
 *
 * Author: Your Full Name <your@email.address>
 * Copyright (c) <year> <copyright holder>
 *
 * This program and the accompanying materials are made available under the
 * terms of the The MIT License which is available at
 * https://opensource.org/licenses/MIT.
 *
 * SPDX-License-Identifier: MIT
 */


var jsupm_sensortemplate = require('jsupm_sensortemplate');

// Create an instance of the sensor
var sensor = new upm_sensortemplate.SensorTemplate(0);

loop();

function loop()
{
    // Call a method from the sensortemplate
    console.log("SensorTemplate says: " + sensor.helloWorld());

    // Call loop every 2 seconds
    setTimeout(loop, 2000);
}

File: examples/c++/sensortemplate.cxx
/*
 * The MIT License (MIT)
 *
 * Author: Your Full Name <your@email.address>
 * Copyright (c) <year> <copyright holder>
 *
 * This program and the accompanying materials are made available under the
 * terms of the The MIT License which is available at
 * https://opensource.org/licenses/MIT.
 *
 * SPDX-License-Identifier: MIT
 */

#include <iostream>
#include <string>

#include "sensortemplate.hpp"
#include "upm_utilities.h"

int
main()
{
    //! [Interesting]
    // Create an instance of SensorTemplate
    upm::SensorTemplate sensor(0);

    while (true) {
        std::cout << "SensorTemplate says: " << sensor.helloWorld() << std::endl;

        // Repeat every 2 seconds
        upm_delay_us(2000000);
    }
    //! [Interesting]
    return 0;
}
